src/effects/native/flangereffect.cpp

```diff
@@ -82,24 +83,30 @@ void FlangerEffect::processChannel(const ChannelHandle& handle,
                                    const EffectProcessor::EnableState enableState,
                                    const GroupFeatureState& groupFeatures) {
     Q_UNUSED(handle);
-    Q_UNUSED(enableState);
-    Q_UNUSED(groupFeatures);
-    Q_UNUSED(sampleRate);
-    CSAMPLE lfoPeriod = m_pPeriodParameter->value();
-    CSAMPLE lfoDepth = m_pDepthParameter->value();
-    // Unused in EngineFlanger
-    // CSAMPLE lfoDelay = m_pDelayParameter ?
-    //         m_pDelayParameter->value().toDouble() : 0.0f;
 
-    // TODO(rryan) check ranges
-    // period needs to be >=0
-    // delay needs to be >=0
-    // depth is ???
+    // TODO: remove assumption of stereo signal
+    const int kChannels = 2;
+
+    // The parameter minimum is zero so the exact center of the knob is 2 beats.
+    double lfoPeriod = m_pPeriodParameter->value();
+    if (groupFeatures.has_beat_length_sec) {
+        // lfoPeriod is a number of beats
+        lfoPeriod = std::max(roundToFraction(lfoPeriod, 2.0), 1/4.0);
+        if (m_pTripletParameter->toBool()) {
+            lfoPeriod /= 3.0;
+        }
+        lfoPeriod = lfoPeriod * groupFeatures.beat_length_sec
+                * sampleRate * kChannels;
+    } else {
+        // lfoPeriod is a number of seconds
```

Comment:
these comments are confusing, because lfoPeriod variable changes unit. We should use two variables for it. 
periodParameter and periodFrames

Reply:
What means lfo by the way?

Reply:
Yes, two variables could make the code easier to understand.

Reply:
Low Frequency Oscillator